operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/informer/InformerConfiguration.java

```diff
@@ -76,10 +138,64 @@ public Set<String> getNamespaces() {
     return namespaces;
   }
 
+  public boolean watchAllNamespaces() {
+    return InformerConfiguration.allNamespacesWatched(getNamespaces());
+  }
+
+  public boolean watchCurrentNamespace() {
+    return InformerConfiguration.currentNamespaceWatched(getNamespaces());
+  }
+
+  public boolean inheritsNamespacesFromController() {
+    return inheritsNamespacesFromController(getNamespaces());
+  }
+
+  /**
+   * Computes the effective namespaces based on the set specified by the user, in particular
+   * retrieves the current namespace from the client when the user specified that they wanted to
+   * watch the current namespace only.
+   *
+   * @return a Set of namespace names the associated controller will watch
+   */
+  public Set<String> getEffectiveNamespaces(ControllerConfiguration<?> controllerConfiguration) {
+    if (inheritsNamespacesFromController()) {
```

Comment:
This should be checked only for informersEventSources not for ControllerEventSource, but I guess we can live with this.

Reply:
Yeah, not great, I agree.